app/controllers/githook_controller.rb

```diff
@@ -13,4 +16,31 @@ def index
       render :nothing => true, :status => 400
     end
   end
+  def dev
+    @success, @output = pull_dev()
+    render :index
+  end
+
+  def master
+    @success, @output = pull_master()
+    render :index
+  end
+
+  def pull_dev
+    out = %x('/var/www/scripts/updatewrapper-dev')
+    return $?.exitstatus, out
+  end
+
+  def pull_master
+    out = %x('/var/www/scripts/updatewrapper')
+    return $?.exitstatus, out
+  end
+
+
+  def authenticate
+    flash[:error] = t('the_role.access_denied')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.